crates/types/src/identifiers.rs

```diff
@@ -1095,13 +1098,16 @@ impl ResourceId for ExternalSignalIdentifier {
 
     /// We use a custom strategy for awakeable identifiers since they need to be encoded as base64
     /// for wider language support.
-    fn push_contents_to_encoder(&self, encoder: &mut IdEncoder<Self>) {
+    fn push_contents_to_encoder<W: fmt::Write>(
+        &self,
+        encoder: &mut IdEncoder<Self, W>,
+    ) -> fmt::Result {
         let mut input_buf =
             BytesMut::with_capacity(size_of::<EncodedInvocationId>() + size_of::<EntryIndex>());
```

Comment:
This can also be replaced by a stack-allocated array instead of the underlying Vec. Same for all the other identifiers. I'll do that as a follow up PR, no need to do that in your PR.